usage/app/lib/UsageRecorder.scala

```diff
@@ -77,20 +86,24 @@ class UsageRecorder(
         val result = func(mediaUsage)
         logger.info(
           logMarker,
-          s"'$opName' DB Operation for ${mediaUsage.grouping} -  on mediaID: ${mediaUsage.mediaId} with result: ${result}"
+          s"'$opName' DB Operation for ${mediaUsage.grouping} - on mediaID: ${mediaUsage.mediaId} with result: $result"
         )
         usageMetrics.incrementUpdated
         result
       }
 
-      val markAsRemovedOps = (dbUsages diff usageGroup.usages)
+      val toMarkAsRemoved = (dbUsageKeys diff streamUsageKeys).flatMap(dbUsageMap.get)
+      val markAsRemovedOps = toMarkAsRemoved
         .map(performAndLogDBOperation(usageTable.markAsRemoved, "markAsRemoved"))
 
-      val createOps = (if(usageGroup.isReindex) usageGroup.usages else usageGroup.usages diff dbUsages)
-        .map(performAndLogDBOperation(usageTable.create, "create"))
+      val toCreate = (if(usageGroup.isReindex) streamUsageKeys else streamUsageKeys diff dbUsageKeys)
+        .flatMap(streamUsageMap.get)
+      val createOps = toCreate.map(performAndLogDBOperation(usageTable.create, "create"))
 
-      val updateOps = (if(usageGroup.isReindex) Set() else usageGroup.usages intersect dbUsages)
-        .map(performAndLogDBOperation(usageTable.update, "update"))
+      val toUpdate = (if (usageGroup.isReindex) Set() else streamUsageKeys intersect dbUsageKeys)
+        .flatMap(streamUsageMap.get)
+        .diff(dbUsages) // to avoid updating to exactly the same data that's already in the DB
```

Comment:
I think the word "exactly" in the comment is a little confusing, since this `diff` is also avoiding data with different dates being written

Reply:
fair point, removed in https://github.com/guardian/grid/pull/3735/commits/4f88a93f12f955708569ea579eab4fade1a57d6e